java/com/google/flatbuffers/FlexBuffers.java

```diff
@@ -483,13 +494,13 @@ public String asString() {
             if (isString()) {
                 int start = indirect(bb, end, parentWidth);
                 int size = readInt(bb, start - byteWidth, byteWidth);
-                return Utf8.getDefault().decodeUtf8(bb, start, size);
+                return bb.getString(start, size);
             }
             else if (isKey()){
                 int start = indirect(bb, end, byteWidth);
                 for (int i = start; ; i++) {
```

Comment:
@aardappel  The decoder should be able to find the end of the string when decoding.

Reply:
Which of my comments is this in reference to? If this is about checking for 0, my point is that that should ONLY be used for keys, not for strings or blobs.

Reply:
Exactly. For example:
`Utf8.getDefault().decodeUtf8(bb, start, size)` for blobs and strings.
`Utf8.getDefault().decodeUtf8NullTerminated(bb, start)` for keys.

Reply:
blobs don't necessarily contain UTF-8